### DISCUSSION TITLE
Midi Input Eement (CC messages)

### DISCUSSION THREAD

**Discussion by FotisDovas** [tttapa/Control-Surface#600] [Q&A]
Hi there! 

Great library! 

I am a flutter dev. So my programming skills are dependant more on intuition, rather than deep coding knowledge.

Thing is, i want to implement a single-LED display, dimming it in accordance with the incoming CC message value, of a certain midi channel. The input is coming from my plugin's, - "dB.toMidiCC" - , message values.

I already captured the incoming midi messages on the serial monitor. But for now, i cannot link the input message to any LED class.

I really want to avoid the VUmeter class, since it binds to some other midi message types, which i want to use for other reasons 
(eg. ultra smooth CC control, of DAW variables [for audio spectrum filters etc])

Digging for a while, i found that there used to be a CCValueLEDPWM class, but maybe it is depracated.
For now, i ll keep studing for a solution, but I had to post this, since I am freaking out a bit : D

So the questions are,
1) Why is the pre-mentioned class no longer in your really CHARMING library ?
2) Is there a way to link leds to CC midi messages ?  (an example would be instant success for me) 
Note leds and Mackie VU don't really make my whole day ( i am also something of a UX designer and have to stick to the concept! ) 

I hope this finds you well and vibing!

**Comment by tttapa** [comment]
You can find a full overview here: https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/de4/group__midi-input-elements-leds.html   
It includes the `CCLEDPWM` class. The reason for the name change from `CCValueLEDPWM` to ` CCLEDPWM` is that it doesn't store the actual value, it only drives the LED directly.

**Reply by FotisDovas** [reply]
I actually admire how soon you reacted! 

I'll get to your suggestion right away!

**Reply by FotisDovas** [reply]
it works perfectly !!!

Excuse me for abusing this channel.. but I will go one step further if you may!

So, I also have a plugin that detects the frequency range of a certain mixer track, and also outputs it, and/or the center of energy (frequency-wise) as Midi CC. 
By the way, the plugins come from the Blue Cat brand.

1) Would you say there is a premade class that could receive up to 2 midi messages from the DAW, and also have getters for the values ?
Then, i could output it as RGB PWM values of an RGB led, in accordance with some conditions-driven code i write?

2) If there is no such premade class, would you say i could create one using the Midi_Input_Element class as a core ?

Thank you in advance! Feel free to ignore me if I am asking for too much. Peace!

**Reply by tttapa** [reply]
> Would you say there is a premade class that could receive up to 2 midi messages from the DAW, and also have getters for the values ?

I'm not sure what you mean by “up to 2”, do you mean in absolute terms, as in “match the first two CC messages that are received ever? Or listening to two specific addresses?

In case of the latter, you can use two instances of the `CCValue` class.  
If not, it might be easiest to just use the MIDI input callbacks, see https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/df7/midi-tutorial.html#midi_md-receiving.

**Reply by FotisDovas** [reply]
Yeah, it is defenitetely the second case! Sorry for not making it clearer, I sometimes talk too abstractly :P 

Sorry for the long delay.
I just received the RGB leds from the post office, and may start experimenting really soon!

The concept in my head is something like: get value of stream from address one, get value of stream from adress two, plot them in some way, as to output peak velocity blinks of certain color, color defined by critical frequency of audio channel (gotten from the CC message output of the plugin).

As for the MIDI input callbacks, i got the idea that they would be used for more general cases, or should i say global.
Anyway, something 'non-targeted'.

will be sending feedback : )

**Reply by FotisDovas** [reply]
It works perfectly, as intended! 
I kinda wish i had ordered HSB Leds though : D xaxa
Thank you man
